### DISCUSSION TITLE
How to change request in route acting as a simple reverse proxy returning the response?

### DISCUSSION THREAD

**Discussion by frederikhors** [actix/actix-web#2832] [Q&A] [answered]
I need to add to my `actix-web` instance a route that acts as a simple reverse proxy according to these criteria:

1. I get a call for `GET https://host_1:8000/some_route` (please don't mind the HTTPS, I'm not handling SSL in my code)

1. the reverse proxy (actix-web route) changes it to: `http://host_2:1234/some_route`

1. the reverse proxy (actix-web route) returns the response

In Go I'm using this simple code:

```go
reverseProxy := &httputil.ReverseProxy{Director: func(r *http.Request) {
  r.URL.Scheme = "http"
  r.URL.Host = "host_2:1234"
}}

router.Group("/some_route", func(group *Group) {
  group.GET("*", reverseProxy)
})
```

In Rust I don't understand how to do:

```rust
#[actix_web::main]
async fn main() -> std::io::Result<()> {
    HttpServer::new(|| {
        App::new()
            .service(hello)
            .route("/some_route", web::get().wrap(SOME_MIDDLEWARE_HERE?).to(WHAT_HERE?)) // This is the line
    })
    .bind(("127.0.0.1", 8000))?
    .run()
    .await
}
```

**Comment by robjtede** [comment, marked as answer]
It's more complicated in Rust of course because there's no built in reverse proxy tools.

We have a somewhat naiive example here though: https://github.com/actix/examples/blob/master/http-proxy/src/main.rs

**Reply by frederikhors** [reply]
Thanks. Do we necessarily need `awc::Client`?

**Reply by robjtede** [reply]
Again, it'd be more complex with reqwest. Though it is strictly a better client atm.

**Reply by frederikhors** [reply]
@robjtede can you help me with: https://github.com/actix/actix-web/discussions/2847.

It comes from your hint of using that and it works but not with the body, why?
